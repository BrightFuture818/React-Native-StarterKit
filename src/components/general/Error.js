/**
 * Error Screen
 *
    <Error text={'Server is down'} />
 *
 * React Native Starter App
 * https://github.com/mcnamee/react-native-starter-app
 */
import React, { PropTypes } from 'react';
import { View } from 'react-native';
import Icon from 'react-native-vector-icons/Ionicons';

// Consts and Libs
import { AppStyles } from '@theme/';

// Components
import { Spacer, Text } from '@ui/';

/* Component ==================================================================== */
const Error = ({ text }) => (
  <View style={[AppStyles.container, AppStyles.containerCentered]}>
    <Icon name={'ios-alert-outline'} size={50} color={'#CCC'} />

    <Spacer size={10} />

    <Text>{text}</Text>
  </View>
);

Error.propTypes = { text: PropTypes.string };
Error.defaultProps = { text: 'Woops, Something went wrong.' };
Error.componentName = 'Error';

/* Export Component ==================================================================== */
export default Error;
