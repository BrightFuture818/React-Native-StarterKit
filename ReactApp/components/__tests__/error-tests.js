/**
 * Test to check if the component renderes correctly
 */
'use strict'

import 'react-native'
import React from 'react'
import Error from '../error'
import renderer from 'react-test-renderer'

/**
 * Check if alerts renders correcly
 * and asserting it to the matching snapshot
 */
it('Error renders correcly', () => {

  const tree = renderer.create(
    <Error />
  ).toJSON()

  expect(tree).toMatchSnapshot()

});
