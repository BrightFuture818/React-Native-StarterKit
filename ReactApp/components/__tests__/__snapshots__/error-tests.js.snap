exports[`test Error renders correcly 1`] = `
<View
  style={
    Array [
      Object {
        "backgroundColor": "#E9EBEE",
        "flex": 1,
        "flexDirection": "column",
        "position": "relative",
      },
      Object {
        "alignItems": "center",
        "justifyContent": "center",
      },
    ]
  }>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#CCC",
          "fontSize": 50,
        },
        undefined,
        Object {
          "fontFamily": "Ionicons",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }>
    ï„ƒ
  </Text>
  <View
    style={
      Array [
        Object {
          "height": 1,
          "left": 0,
          "marginTop": 10,
          "right": 0,
        },
      ]
    } />
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#222222",
          "fontFamily": "Avenir",
          "fontSize": 14,
          "fontWeight": "500",
          "lineHeight": 21,
        },
      ]
    }>
    Woops, Something went wrong.
  </Text>
</View>
`;
